openstef/model/metamodels/grouped_regressor.py

```diff
@@ -0,0 +1,214 @@
+                )
+            )
+        for c in self.group_columns:
+            if c not in list(df.columns):
+                raise ValueError("The group column {} is missing!".format(c))
+
+    def _partial_fit(
+        self, group: Any, df_group: pd.DataFrame, eval_set=None, **kwargs
+    ) -> Tuple[Any, BaseEstimator]:
+        estimator = clone(self.base_estimator)
+        X = df_group.loc[:, self.feature_names_]
+        y = df_group.loc[:, "__target__"]
+
+        if eval_set is None:
+            estimator_fitted = estimator.fit(X, y, **kwargs)
+        else:
+            estimator_fitted = estimator.fit(
+                X,
+                y,
+                eval_set=[
+                    (
+                        df.loc[
+                            (df[self.group_columns] == group).to_numpy().flatten(),
+                            self.feature_names_,
+                        ],
+                        df.loc[
+                            (df[self.group_columns] == group).to_numpy().flatten(),
+                            "__target__",
+                        ],
+                    )
+                    for df in eval_set
+                ],
+                **kwargs
+            )
+
+        return (group, estimator_fitted)
+
+    def _partial_predict(self, group, df_group, **kwargs):
+        return self.estimators_[group].predict(df_group, **kwargs)
+
+    @classmethod
+    def grouped_compute(
+        cls,
+        df: pd.DataFrame,
+        group_columns: Union[List[str], List[int]],
+        func: Callable[[Tuple, pd.DataFrame], np.array],
+        n_jobs: int = 1,
+        eval_set=None,
+    ) -> Tuple[Tuple[np.array, ...], DataFrameGroupBy, pd.DataFrame]:
+        """Computes the specified function on each group defined by the grouping columns.
+        It is an utility function used to perform fit and predict on each group.
+        The df_res is the final dataframe that aggregate the results for each group.
+        The group_res is a tuple where each field is corresponding to a results for a group.
+        The gb is the grouping object.
+
+        Args:
+            df: (pd.DataFrame) data frame containing the input data necessary for the computation .
+            group_columns: (list) list of the columns used for the groupby operation
+            func: (Callable) function that take the group key and the conrresponding data of this group
+             and perform the computation on this group.
+            n_jobs: (int) The maximum number of concurrently running jobs,
+
+        Returns:
+            results (tuple): the tuple of the results of each group, the grouping dataframe and the global dataframe of results.
+        """
+
+        index_name = df.index.name or "index"
+        df_reset = df.reset_index()
+
+        df_res = pd.DataFrame(index=df_reset.index)
+
+        gb = df_reset.groupby(group_columns)
+
+        if n_jobs > 1:
+            group_res = joblib.Parallel(n_jobs=n_jobs)(
```

Comment:
Preferably parallel scaling is done at a higher level (e.g. cluster level with unlimited multiple pods/jobs that run openstef jobs simultaneously in a kubernetes cluster) instead of at process level, that is often limited by amount of available CPU processors. Scaling at both levels (cluster + process) becomes complex too since it is unclear what is scaled when. This function is okay now, but let's add a comment that preferred parallel scaling is at cluster level instead of process level
```suggestion
            # Preferred scaling is at cluster level (e.g. k8s/serverless) instead of process level
            group_res = joblib.Parallel(n_jobs=n_jobs)(
```

Reply:
You are right, we could even log a warning to be sure that it is at the risk of the user. However my idea is that we can keep it, if someone use it out of the box the parallel scaling could be useful.